### ISSUE TITLE
naming themes' template files

### ISSUE THREAD

**Issue by KingArchee** [GetSimpleCMS/GetSimpleCMS#1204]
Giving specific theme templates a name, like "Default Template" for _template.php_

**Comment by tablatronix** [comment]
but how...
A filename mapping ?

suggestions ?

**Comment by webketje** [comment]
What about (dot) `.template.php`? The Default Template is `template.php`, and every other theme file named e.g. `landing.template.php` or `contact.template.php` maps to the "subdomain" with as unslugified-capitalized. (e.g. `fluid-grid.template.php => Fluid Grid`)

**Comment by tablatronix** [comment]
ok I think themes should provide a definition file, I do not think a naming convention is even necessary. Unless you see a need for it.

We can add some defaults for starters.

**Comment by tablatronix** [comment]
suggestions ?

- blog
- gallery
- sidebar (left/right/etc)

**Comment by webketje** [comment]
I would rather say "I do not think a definition file is even necessary. Unless you see a need for it.". Imo a definition file is more friction than using the naming convention, and the latter is readable simply in the dir structure. Plugins like i18n also use _lang at the end of a .xml file. And existing default theme includes already use the '.inc' pattern.

**Comment by tablatronix** [comment]
I do not find filenames to be dynamic enough, nor do they support i18n. Plus trying to auto parse a filename to natural name is a pain in the ass, spaces, underscores, utf_8, uppercase rules ?

Definition is not necessary for the builtin ones, only additional ones.

I am not in favor of using filenames as natural names, seems kludgy

**Comment by lokothodida** [comment]
> @tablatronix : ok I think themes should provide a definition file

So something like a theme manifest file? :

```xml
<?xml version="1.0" encoding="UTF-8"?>
<theme>
  <name>Innovation</name>
  <version>1.0</version>
  <author>Chris Cagle</author>
  <templates>
    <template file="template.php">Default Template</template>
    <template file="sidebar.inc.php">Sidebar</template>
  </templates>
</theme>
```

**Sidenote**: Maybe even allow theme developers to specify theme-relative settings directly in the manifest.

Is there a current mechanism/standard for handling language files in themes? If it is the same as plugins (so a `lang` directory in the theme's folder with PHP language files per translation), then maybe allow i18n strings with a `@` prefix:

```xml
    <template file="template.php">@Innovation/DEFAULT_TEMPLATE</template>
    <template file="sidebar.inc.php">@Innovation/SIDEBAR</template>
```

**Comment by tablatronix** [comment]
yeah this has been discussed elsewhere also, not sure where ....
but we could eventually use this for declaring theme runtime components/snippets,admin css or other weird stuff

**Comment by tablatronix** [comment]
Where does @ notation come from is that just a novel token idea, or just a language namespace ( do we already have these for plugins do we not? )

**Comment by lokothodida** [comment]
`@` is the delimiter used by [Android manifests](https://developer.android.com/training/basics/supporting-devices/languages.html) for localization, which are also XML files, although it gets used mainly for attributes rather than node contents. So the above example would instead be:

```xml
<template file="template.php" name="@Innovation/DEFAULT_TEMPLATE" />
<template file="sidebar.php" name="@Innovation/SIDEBAR" />
```

**Comment by hameau** [comment]
I agree with the basic idea (naming theme files), but trying to produce names algorithmically seems to be over-complicating things.

Why not just have a tag in the template file? Something like this:
```
/****************************************************
*
* @File:		template.php
* @Package:		GetSimple
* @Action:		Cardinal theme for GetSimple CMS
* @TemplateName:	Default Template
*
*****************************************************/
```
If the ```@TemplateName``` is missing, fall back to the filename. Easy, flexible, optional and backwards-compatible.

**Comment by tablatronix** [comment]
Then you have use a custom parser and load all files

**Comment by hameau** [comment]
> Then you have use a custom parser and load all files

Fair point.

However, an independent manifest would be a complete ballache during site development, unless it can be disabled and fall back to a file list.

**Comment by tablatronix** [comment]
The manifest is just to enhance with natural names. Filenames will always work
